ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerFragment.kt

```diff
@@ -68,11 +68,27 @@ class HomeScreenMapContainerFragment : AbstractMapContainerFragment() {
       .`as`(RxAutoDispose.autoDisposable(this))
       .subscribe { onZoomThresholdCrossed() }
 
-    adapter = LoiCardAdapter()
-    adapter.setLoiCardFocusedListener { mapFragment.setActiveLocationOfInterest(it) }
+    adapter = MapCardAdapter()
+    adapter.setLoiCardFocusedListener {
+      if (it?.loiId != null) {
+        mapFragment.setActiveLocationOfInterest(it.loiId)
+      }
```

Comment:
Maybe use `?.let{}`

Reply:
Updated this now that I decided to use a sealed class for MapCardUiData